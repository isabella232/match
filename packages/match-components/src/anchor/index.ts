export * from "./anchor";
export * from "./constants";
export * from "./types";
