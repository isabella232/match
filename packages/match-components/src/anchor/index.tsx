import * as React from "react";
import * as PropTypes from "prop-types";
import { StyledAnchor } from "./styles";
import { AnchorVariant, AnchorProps } from "./types";

export interface ExternalProps {
  rel: string | undefined;
  target: string | undefined;
}

const EXTERNAL_URL_REGEX = /^(https?:)\S*$/;

const secureExternalLink = (href: string | undefined): ExternalProps => {
  if (href) {
    if (EXTERNAL_URL_REGEX.test(href)) {
      return { rel: "noreferrer noopener", target: "_blank" };
    }
  }
  return { rel: undefined, target: undefined };
};

const Anchor: React.FC<AnchorProps> = ({ ...props }) => {
  if (!props.href) console.warn("href must be provided for an anchor");
  const externalURL = secureExternalLink(props.href);
  return (
    <StyledAnchor
      {...props}
      rel={props.rel ? props.rel : externalURL.rel}
      target={props.target ? props.target : externalURL.target}
    />
  );
};

Anchor.propTypes = {
  variant: PropTypes.oneOf(Object.values(AnchorVariant)),
  href: PropTypes.string,
  target: PropTypes.string,
  rel: PropTypes.string,
  noUnderline: PropTypes.bool,
};

Anchor.defaultProps = {
  variant: AnchorVariant.TEXT,
  noUnderline: false,
};

export { Anchor, AnchorVariant, AnchorProps };
