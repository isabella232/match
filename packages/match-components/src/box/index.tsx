import * as React from "react";
import { spaceTokenProp } from "@twilio-labs/match-props";
import { BoxProps } from "./types";
import { StyledBox } from "./styles";

const Box: React.FC<BoxProps> = ({ ...props }) => {
  return <StyledBox {...props} />;
};

Box.propTypes = {
  padding: spaceTokenProp,
  paddingLeft: spaceTokenProp,
  paddingRight: spaceTokenProp,
  paddingBottom: spaceTokenProp,
  paddingTop: spaceTokenProp,
  margin: spaceTokenProp,
  marginLeft: spaceTokenProp,
  marginRight: spaceTokenProp,
  marginBottom: spaceTokenProp,
  marginTop: spaceTokenProp,
};

export { Box, BoxProps };
