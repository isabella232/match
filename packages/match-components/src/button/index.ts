export * from "./constants";
export * from "./button";
export * from "./types";
