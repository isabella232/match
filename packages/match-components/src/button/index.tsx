import * as React from "react";
import * as PropTypes from "prop-types";
import { uid } from "react-uid";
import { StyledButton, StyledPrompt } from "./styles";
import { ButtonVariant, ButtonType, ButtonSize } from "./types";
import type { ButtonProps } from "./types";

const Button: React.FC<ButtonProps> = ({
  type,
  prompt,
  children,
  ...props
}) => {
  if (props.download && !props.href)
    console.warn("[Button]: Href must be provided for a download link.");

  return (
    <StyledButton
      forwardedAs={props.href ? "a" : undefined}
      type={!props.href ? type : undefined}
      {...props}
    >
      {Array.isArray(children)
        ? children.map((child) => {
            if (React.isValidElement(child))
              return React.cloneElement(child, { key: uid(child) });
            if (typeof child === "string") return child.trim();
          })
        : children}
      {prompt && <StyledPrompt />}
    </StyledButton>
  );
};

Button.propTypes = {
  variant: PropTypes.oneOf(Object.values(ButtonVariant)),
  type: PropTypes.oneOf(Object.values(ButtonType)),
  size: PropTypes.oneOf(Object.values(ButtonSize)),
  href: PropTypes.string,
  disabled: PropTypes.bool,
  fullWidth: PropTypes.bool,
  download: PropTypes.bool,
  prompt: PropTypes.bool,
};

Button.defaultProps = {
  variant: ButtonVariant.PRIMARY,
  type: ButtonType.BUTTON,
  size: ButtonSize.NORMAL,
  disabled: false,
  fullWidth: false,
  download: false,
  prompt: false,
};

Button.displayName = "Button";

export { Button, ButtonVariant, ButtonType, ButtonSize };
export type { ButtonProps };
