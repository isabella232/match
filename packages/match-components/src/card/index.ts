export * from "./card";
export * from "./constants";
export * from "./types";
