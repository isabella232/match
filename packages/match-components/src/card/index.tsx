import * as React from "react";
import { spaceTokenProp } from "@twilio-labs/match-props";
import { CardProps, CardVariant } from "./types";
import { StyledCard } from "./styles";
import * as PropTypes from "prop-types";

const Card: React.FunctionComponent<CardProps> = ({ ...props }) => {
  return <StyledCard {...props} />;
};

Card.propTypes = {
  variant: PropTypes.oneOf(Object.values(CardVariant)),
  padding: spaceTokenProp,
  paddingLeft: spaceTokenProp,
  paddingRight: spaceTokenProp,
  paddingBottom: spaceTokenProp,
  paddingTop: spaceTokenProp,
  margin: spaceTokenProp,
  marginLeft: spaceTokenProp,
  marginRight: spaceTokenProp,
  marginBottom: spaceTokenProp,
  marginTop: spaceTokenProp,
};

Card.defaultProps = {
  variant: CardVariant.PRIMARY,
};

export { Card, CardProps };
