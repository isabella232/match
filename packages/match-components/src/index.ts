export * from "./wombat";
export * from "./visually-hidden";
export * from "./button";
