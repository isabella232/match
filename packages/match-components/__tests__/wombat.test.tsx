import * as React from "react";
import * as renderer from "react-test-renderer";
import { render } from "react-dom";
import { axe } from "jest-axe";
import { Wombat } from "../src";

describe("Wombat", () => {
  it("it should render Wombat with the brand color", (): void => {
    const tree = renderer
      .create(<Wombat droppings={["🌯", "🍦", "🍔"]} color="brand" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    const container = document.createElement("div");
    container.setAttribute("role", "main");
    document.body.append(container);
    render(<Wombat droppings={["🌯", "🍦", "🍔"]} color="brand" />, container);
    const results = await axe(document.body);
    expect(results).toHaveNoViolations();
  });
});
