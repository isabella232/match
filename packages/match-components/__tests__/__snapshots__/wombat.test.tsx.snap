// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wombat it should render Wombat with the brand color 1`] = `
<DocumentFragment>
  .c0 {
  color: hsla(217.33333333333334,42.05607476635513%,20.980392156862745%,1);
  font-weight: 400;
  font-size: 1rem;
  font-family: Whitney SSm A,Whitney SSm B,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c1 {
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(3,max-content);
}

.c1 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 2rem;
  height: 2rem;
  background: hsla(231.24087591240877,100%,26.862745098039216%,1);
  border-radius: 0.5rem;
}

<div
    class="c0"
  >
    <div
      class="c1"
      color="brand"
    >
      <div>
        🌯
      </div>
      <div>
        🍦
      </div>
      <div>
        🍔
      </div>
    </div>
  </div>
</DocumentFragment>
`;
